pylds/discrete_maps.py

```diff
@@ -0,0 +1,130 @@
+
+import numpy as np
+
+def StandardMap(u_initial, PARAMETERS=[0.3]):
+    """
+    Returns 1D Hamilton-Centre vector field at time t, for an array of points in phase space.
+    Number of model parameters: 1 . PARAMETERS = [omega]
+    Functional form: v = (omega*y, - omega*x), with u = (x, y)
+    
+    Parameters
+    ----------
+    u : array_like, shape(n,)
+        points in phase space to determine vector field at time t
+    
+    PARAMETERS : list of floats
+        vector field parameters
+    
+    Returns
+    -------
+    v : array_like, shape(n,)
+        vector field corresponding to points u, in phase space at time t
+    """
+    x_initial, y_initial = u_initial.T
+    # Hamiltonian Model Parameter
+    K, = PARAMETERS
+    
+    # Map components 
+    x_next = x_initial + y_initial - (K/(2*np.pi))*np.sin(2*np.pi*x_initial)
+    y_next = y_initial - (K/(2*np.pi))*np.sin(2*np.pi*x_initial)
```

Comment:
The Standard map is defined on a square with periodic boundary condition. This function maps point from outside [-0.5,0.5] into the interval, creating integer jumps in the values of Lagrangian descriptors, hence not a periodic output.

Reply:
Check grid dimensions to see jumps due to absence of periodicity.

Reply:
![download](https://user-images.githubusercontent.com/52881162/96131040-9f26eb80-0ef0-11eb-85b0-4795b9cf0a49.png)

Reply:
Can't be right

Reply:
Yep, working on it.

Reply:
It should be enough to take x mod 1 and p_x mod 1 at the very beginning. It's going to be more efficient than doing so after every step.

Also there is a discrepancy between the implemented equation and those in [Chirikov's original paper](http://www.quantware.ups-tlse.fr/chirikov/refs/chi1969e.pdf): there is a plus (not minus) in front of K and substituting `y_next` into `x_next`, `np.sin` is called only once. It is crucial that the package produces output comparable to existing work, which doesn't seem to be the case now. Compare the figure below with figures in [Chirikov's scholarpedia article](http://www.scholarpedia.org/article/Chirikov_standard_map).
![download](https://user-images.githubusercontent.com/52881162/96241460-18c6e400-0f9a-11eb-8f6c-360b7248fd57.png)

Reply:
Commit 043d653 gives the same result. The jumps come from the difference between the initial conditions and their first images. Input should be taken mod 1, if you want to go down this route

Reply:
Computation of the LDs using Meiss definition of the standard map without PBCs (unwrapped trajectories) with initial conditions in the domain [-1/2, 1/2]x[-1/2, 1/2] (periodic cell) leads to results that match his, for K=1 and 2. See figures below.

The caveat. If you extend the dimensions of the visualization slide the LD outputs clearly won't be periodic. To do so, you need to redefine the standard map and also modify the `lagrangian_descriptor` function to consider the visualization cell to compute the correct distance between subsequent points (distance in a torus or cylinder). However, the periodic output won't be any different to the output generated for a visualization slide matching the periodic cell dimensions.

K = 1
![image](https://user-images.githubusercontent.com/17453074/97579029-7bcc6800-19e9-11eb-8079-a5cb385f79fd.png)
![image](https://user-images.githubusercontent.com/17453074/97579052-8129b280-19e9-11eb-8d65-78e6532decfd.png)
![image](https://user-images.githubusercontent.com/17453074/97579066-8686fd00-19e9-11eb-9879-3df3f1eb3aa0.png)

K = 2
![image](https://user-images.githubusercontent.com/17453074/97579149-9dc5ea80-19e9-11eb-9669-11446cabf9bc.png)
![image](https://user-images.githubusercontent.com/17453074/97579164-a3bbcb80-19e9-11eb-9620-baa3454f9538.png)
![image](https://user-images.githubusercontent.com/17453074/97579180-a8807f80-19e9-11eb-8be1-6300613551d2.png)

Compare with the original Figure from paper. 
![image](https://user-images.githubusercontent.com/17453074/97579254-b9c98c00-19e9-11eb-838f-6b3eb95ab44b.png)

Reply:
Add function to move input visualisation slice to unit square centred at origin.

Reply:
Add function to redefine user inputed grid dimensions to unit square centred at origin.

Reply:
> Add function to redefine user inputed grid dimensions to unit square centred at origin.

Needless.

Applied PBC with minimum image criterion to displacements `dy = f(y) - y0`. See new lines in `compute_lagrangian_descriptor`.

This solves the issue. Now, one needs to pass the length of the unit cell with PBCs to `compute_lagrangian_descriptor` so that the LD output is periodic as expected.

K = 1
![image](https://user-images.githubusercontent.com/17453074/99063398-5bc9a680-259c-11eb-9641-64f434c982a7.png)

K=2
![image](https://user-images.githubusercontent.com/17453074/99063514-8582cd80-259c-11eb-9d44-7593850b6516.png)